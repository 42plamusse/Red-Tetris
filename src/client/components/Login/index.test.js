import React from "react";
import thunk from "redux-thunk";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import { Login } from "./index";

// const mockStore = configureStore([thunk]);
// const initialState = {
//   playerName: "",
//   currentRoom: ""
// };
// const store = mockStore(initialState);
// const props = {
//   newPlayer: jest.fn()
// };

const props = {
  newPlayer: jest.fn()
};

const wrapper = shallow(<Login {...props} />);
console.log(toJson(wrapper));

describe("<Login />", () => {
  test("test Login rendering without crashing", () => {
    const component = wrapper;

    expect(toJson(component)).toMatchSnapshot();
  });
  // test("newPlayer action", () => {
  //   const component = wrapper;

  //   component.props().newPlayer("test");
  //   expect(store.getActions()).toMatchSnapshot();
  // });
});
