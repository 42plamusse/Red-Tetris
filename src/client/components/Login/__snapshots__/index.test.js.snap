// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> test Login rendering without crashing 1`] = `
<LoginSub
  error={
    Object {
      "boolean": null,
      "message": "",
    }
  }
  label="Name"
  onChange={[Function]}
  onSubmit={[Function]}
  value=""
/>
`;
